src/core/app/controllers/anthropic_controller.py

```
@@ -533,9 +533,32 @@ def get_anthropic_controller(service_provider: IServiceProvider) -> AnthropicCon
                     agent_response_formatter = AgentResponseFormatter()
 
                 session_manager = SessionManager(session, session_resolver)
-                backend_request_manager = BackendRequestManager(
-                    backend_processor, response_proc
+                backend_request_manager_service = service_provider.get_service(
+                    cast(type, IBackendRequestManager)
```

Ruff (0.13.3)

Undefined name `IBackendRequestManager`

(F821)